src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -86,68 +86,127 @@ public String getScript() {
         // This causes the Jenkins output to contain bogus characters because Java does not handle the BOM characters by default.
         // This code mimics Out-File, but does not write a BOM.  Hopefully PowerShell will provide a non-BOM option for writing files in future releases.
-        String helperScript = "Function Out-FileNoBom {\n" +
-        "[CmdletBinding()]\n" +
-        "param(\n" +
-        "  [Parameter(Mandatory=$true, Position=0)] [string] $FilePath,\n" +
-        "  [Parameter(ValueFromPipeline=$true)] $InputObject\n" +
-        ")\n" +
-        "  $out = New-Object IO.StreamWriter $FilePath, $false\n" +
-        "  try {\n" +
-        "    $Input | Out-String -Stream | % { $out.WriteLine($_) }\n" +
-        "  } finally {\n" +
-        "    $out.Dispose()\n" +
-        "  }\n" +
-        "}\n" +
-        "Function Execute-AndWriteOutput {\n" +
-        "[CmdletBinding()]\n" +
-        "param(\n" +
-        "  [Parameter(Mandatory=$true)]  [string]$MainScript,\n" +
-        "  [Parameter(Mandatory=$false)] [string]$OutputFile,\n" +
-        "  [Parameter(Mandatory=$true)]  [string]$LogFile,\n" +
-        "  [Parameter(Mandatory=$true)]  [string]$ResultFile,\n" +
-        "  [Parameter(Mandatory=$false)] [switch]$CaptureOutput\n" +
-        ")\n" +
-        "  if ($CaptureOutput -eq $true) {\n" +
-        "      if ($PSVersionTable.PSVersion.Major -ge 5) {\n" +
-        "          $(& $MainScript | Out-FileNoBom -FilePath $OutputFile) 2>&1 3>&1 4>&1 5>&1 6>&1 | Out-FileNoBom -FilePath $LogFile; $LastExitCode | Out-File -FilePath $ResultFile -Encoding ASCII;\n" +
-        "      } else {\n" +
-        "          $(& $MainScript | Out-FileNoBom -FilePath $OutputFile) 2>&1 3>&1 4>&1 5>&1 | Out-FileNoBom -FilePath $LogFile; $LastExitCode | Out-File -FilePath $ResultFile -Encoding ASCII;\n" +
-        "      }\n" +
-        "  } else {\n" +
-        "      & $MainScript *>&1 | Out-FileNoBom -FilePath $LogFile; $LastExitCode | Out-File -FilePath $ResultFile -Encoding ASCII;\n" +
-        "  }\n" +
-        "}";
-        
-        // Execute the script, and catch any errors in order to properly set the jenkins build status. $LastExitCode cannot be solely responsible for determining build status because
-        // there are several instances in which it is not set, e.g. thrown exceptions, and errors that aren't caused by native executables.
-        String wrapperScriptContent = "try {\r\n" + 
-        "  & '" + quote(c.getPowerShellScriptFile(ws)) + "'\r\n" + 
-        "} catch {\r\n" + 
-        "  Write-Error $_;" + 
-        "  exit 1;\r\n" + 
-        "} finally {\r\n" +
-        "  if ($LastExitCode -ne $null) {\r\n" + 
-        "    exit $LastExitCode;\r\n" +
-        "  } elseif ($error.Count -gt 0 -or !$?) {\r\n" + 
-        "    exit 1;\r\n" +
-        "  } else {\r\n" +
-        "    exit 0;\r\n" +
+        String helperScript = "function New-StreamWriter {\r\n" +
+        "[CmdletBinding()]\r\n" +
```

Comment:
AFAICT this is a single string constant, so it would be more readable and maintainable to move it to `src/main/resources/org/jenkinsci/plugins/durabletask/powershellHelper.ps1` and copy it directly from there (`FilePath.copyFrom` with `Class.getResource`) to `getPowerShellHelperFile(ws)`.